README.md

```diff
@@ -83,6 +83,67 @@ npm run prettier
 ngrok http 4000
 ```
 
+## Architecture
+
+#### Frontend
+
+React Application that when loaded uses Socket.io to create a socket connection with the backend API to recieve events from the API. The React Application will be hosted on a seperate EC2 instance. 
+
+#### API
+
+The backend Api do little computation but mainly acts as a carrier between the frontend and storage.
+
+#### Storage
+
+Dynamo DB Tables:
+
+Dynamo Db table can only be queried on their primary key and is ordered by their sort key. So an example query from the messages table could be get all items matching AccountHolderId-AnonymousUserId = 1234-4 and CreatedAt > 23434543 epoch seconds.
+
+Active Users table:
+
+| User Id | Active | Number of Anonymous Users | ... |
+An active sessions table will keep track of each user and whether or not they are active. In the case of the account holder their will be an additional attribute stating the number of anonymous users. For all users in the active sessions table any additional needed attributes such as destination ipaddress or port numbers can go here. The AnonymousUserID is the number of anonymous user that has messaged the Account Holder, if the anonymous user is the 4th person to send a message then the AnonymousUserId will be 4. 
+
+Messages Table:
+
+| AccountHolderId-AnonymousUserId | CreatedAt | Message | PostedBy | ToBeDeletedAt | 
+A messages table will keep track of all of the messages. The messages table is partitioned by a primary key consisting of the AccountHolder Id concatenated with the anonymous user id. This concatenation creates a channel-Id. The messages table will have a sort key on time created and a time to delete attribute for automatic deletion of messages after a certain period. A PostedBy attribute will be set to AccountHolder or Anonymous user to show who sent the message.
+
+
+Room Codes Table 
+
+| Room Code | AccountHolderId |
+A RoomCode table will map room codes to AccountHolders.
```

Comment:
Can't this just be a nullable field in the user table? Genuinely not sure if that's bad practice but if not that seems simpler.